Packs/AnomaliSecurityAnalyticsAlerts/Integrations/AnomaliSecurityAnalyticsAlerts/AnomaliSecurityAnalyticsAlerts.py

```diff
@@ -0,0 +1,343 @@
+                        "to": 1738706820000,
+                        "timezone": "America/New_York"})
+
+        Returns:
+            Response from API.
+        """
+        data = {
+            'query': query,
+            'source': source,
+            'time_range': time_range
+        }
+        return self._http_request(method='POST',
+                                  url_suffix='/api/v1/xdr/search/jobs/',
+                                  json_data=data)
+
+    def get_search_job_status(self, job_id: str) -> dict:
+        """
+        Get the status of a search job.
+
+        Args:
+            job_id: the search job uuid
+
+        Returns:
+            Response from API.
+        """
+        return self._http_request(method='GET',
+                                  url_suffix=f'/api/v1/xdr/search/jobs/{job_id}/')
+
+    def get_search_job_results(self, job_id: str, offset: int = 0, fetch_size: int = 25) -> dict:
+        """
+        Get the results of a search job.
+
+        Args:
+            job_id: the search job uuid
+            offset: offset for pagination. Default is 0.
+            fetch_size: number of records to fetch. Default is 25.
+
+        Returns:
+            Response from API.
+        """
+        params = {'offset': offset, 'fetch_size': fetch_size}
+        return self._http_request(method='GET',
+                                  url_suffix=f'/api/v1/xdr/search/jobs/{job_id}/results/',
+                                  params=params)
+
+    def update_alert(self, data: dict) -> dict:
+        """
+        Update alert data (status or comment).
+
+        Args:
+        data (dict): A dictionary containing the update parameters. It should include:
+            - table_name (str): The name of the table to update (e.g. "alert").
+            - columns (dict): A dictionary mapping column names to their new values.
+            - primary_key_columns: A list of primary key column names.
+            - primary_key_values: A list of lists, where each inner list contains
+              the corresponding values for the primary key columns.
+
+        """
+        return self._http_request(method='PATCH',
+                                  url_suffix='/api/v1/xdr/event/lookup/iceberg/update/',
+                                  json_data=data)
+
+    def check_connection(self) -> dict:
+        """
+        Test connection by retrieving version info from the API.
+        """
+        return self._http_request(method='GET',
+                                  url_suffix='/api/v1/xdr/get_version/')
+
+
+""" COMMAND FUNCTIONS """
+
+
+def command_create_search_job(client: Client, args: dict) -> CommandResults:
+    """Start a search job for IOCs.
+
+    Args:
+        client (Client): Client object with request
+        args (dict): Usually demisto.args()
+
+    Returns:
+        CommandResults.
+    """
+    query = str(args.get('query'))
+    source = str(args.get('source', ''))
```

Comment:
When source is not provided the filter results will be ''? Don't think this correct behavior